recipe-server/normandy/recipes/api/v2/shield_identicon.py

```diff
@@ -0,0 +1,81 @@
+from django.template.loader import render_to_string
+
+
+def pick_pair(genome, base_color):
+    good_pairs = [
+        color for color in genome.colors if base_color.contrasts_well(color)]
+    assert len(good_pairs), 'no colors satisfy the luminance requirement'
+    return genome.choice(good_pairs)
+
+
+def single_color(genome, context):
```

Comment:
Passing in an object that gets modified is a pattern we typically avoid; instead, consider returning an object, and then updating the context with it using `context.update(extra)` in `generate_svg`.

Modifying an argument as a way of returning data isn't recommended because it makes it harder to track where mutations are happening to your data. In this small case it's pretty easy to understand, but broadly applied, the pattern makes code harder to comprehend, so we just avoid using it at all.